packages/protocol/migrations_sol/MigrationL2.s.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity >=0.8.7 <0.8.20;
+
+import { Script } from "forge-std-8/Script.sol";
+import { MigrationsConstants } from "@migrations-sol/constants.sol";
+import "forge-std/console.sol";
+
+import "@celo-contracts/common/FixidityLib.sol";
+import "@celo-contracts-8/common/UsingRegistry.sol";
+
+contract MigrationL2 is Script, MigrationsConstants, UsingRegistry {
+  using FixidityLib for FixidityLib.Fraction;
+
+  /**
+   * Entry point of the script
+   */
+  function run() external {
+    vm.startBroadcast(DEPLOYER_ACCOUNT);
+
+    setupUsingRegistry();
+    activateCeloDistributionSchedule();
+
+    vm.stopBroadcast();
+  }
+
+  function setupUsingRegistry() public {
+    _transferOwnership(DEPLOYER_ACCOUNT);
+    setRegistry(REGISTRY_ADDRESS);
+  }
+
+  function activateCeloDistributionSchedule() public {
+    // TODO: Move this to MigrationsConstants
+    uint256 l2StartTime = 1721909903 - 5; // Arbitrarily 5 seconds before last black
+    FixidityLib.Fraction memory communityRewardFraction = FixidityLib.newFixedFraction(1, 100); // 0.01
+    FixidityLib.Fraction memory carbonOffsettingFraction = FixidityLib.newFixedFraction(1, 1000); // 0.001
+    address carbonOffsettingPartner = 0x22579CA45eE22E2E16dDF72D955D6cf4c767B0eF;
```

Comment:
you can query this from the current chain in EpochRewards contract

Reply:
I think all of them

Reply:
I managed to fetch `communityRewardFraction` and `carbonOffsettingFraction` from the `EpochRewards.sol` contract. I implemented a new interface for this:

https://github.com/celo-org/celo-monorepo/blob/fe05490e15d101d2d4f781408a989f0e6f79937e/packages/protocol/contracts/governance/interfaces/IEpochRewards.sol#L3-L6

But couldn't quite work out how to get the `carbonOffsettingPartner` address, because it's a public variable, and it doesn't have an explicit getter function. I didn't know exactly how to define that in the interface. 

https://github.com/celo-org/celo-monorepo/blob/fe05490e15d101d2d4f781408a989f0e6f79937e/packages/protocol/contracts/governance/EpochRewards.sol#L70-L70

Reply:
you'l have to add the getter to IEpochRewards and implement the function on EpochRewards.

Reply:
Hmm, I'm not sure you'll need to implement a getter in on `EpochRewards.sol`, because public state variables automatically has getter functions: 

> Public state variables differ from internal ones only in that the compiler automatically generates [getter functions](https://docs.soliditylang.org/en/latest/contracts.html#getter-functions) for them, which allows other contracts to read their values.

Source: [docs.soliditylang.org](https://docs.soliditylang.org/en/latest/contracts.html#state-variable-visibility)

I couldn't find a way to reference this "automatic" getter in the `IEpochRewards.sol` interface. Without a matching function in the interface, I wasn't able to call this function. A simple way around this would be to import the implementation (`EpochRewards.sol`) instead of the interface (`IEpochRewards.sol`) since we'll have all contracts on Solidity >0.8. I think the only reason I'm using an interface is because the implementation is on 0.5 and the interface is on 0.8. 

For now, I propose to merge this hard-coded and then a future PR can implement a fix for this if necessary.